var/spack/repos/builtin/packages/py-python-lsp-server/package.py

```diff
@@ -16,15 +16,22 @@ class PyPythonLspServer(PythonPackage):
 
     license("MIT")
 
+    version("1.10.0", sha256="0c9a52dcc16cd0562404d529d50a03372db1ea6fb8dfcc3792b3265441c814f4")
     version("1.7.1", sha256="67473bb301f35434b5fa8b21fc5ed5fac27dc8a8446ccec8bae456af52a0aef6")
     version("1.7.0", sha256="401ce78ea2e98cadd02d94962eb32c92879caabc8055b9a2f36d7ef44acc5435")
     version("1.6.0", sha256="d75cdff9027c4212e5b9e861e9a0219219c8e2c69508d9f24949951dabd0dc1b")
 
     depends_on("python@3.7:", type=("build", "run"))
+    depends_on("python@3.8:", type=("build", "run"), when="@1.8.0:")
     depends_on("py-setuptools@61.2.0:", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("py-setuptools@61.2.0:", type=("build", "run"), when="@:1.7")
```
Otherwise this will be applied to all versions